kurobako_core/src/problem.rs

```diff
@@ -64,6 +66,12 @@ impl ProblemSpecBuilder {
         self
     }
 
+    /// Sets the value reference point of this problem.
+    pub fn reference_point(mut self, reference_point: Option<Params>) -> Self {
+        self.reference_point = reference_point;
+        self
+    }
+
     /// Builds a `ProblemSpec` with the given settings.
     pub fn finish(self) -> Result<ProblemSpec> {
```

Comment:
I think that we should check whether the dimension of `self.values` matches the dimension of `self.reference_point` here (if the latter isn't `None`).

Reply:
Makes sense. Defined a private function `validate` returning a `Result<()>` in https://github.com/sile/kurobako/pull/33/commits/84ee3f22c28f4ee5c9ea9814b2f58e89d99bad1d. I'm not so sure whether I'm doing something ugly or anti-pattern. Glad for any sorts of feedback. Thanks.